scripts/logger_pc_app/user_logger.py

```diff
@@ -0,0 +1,61 @@
+from enum import Enum
+import socket
+import time
+import threading
+import queue
+
+class LogLevel(Enum):
+    ERROR = 0
+    WARN = 1
+    INFO = 2
+    DEBUG = 3
+
+class UserLogger:
+    def __init__(self, module_id, stdout_queue, socket_writer_queue, socket_status_flag):
+        self.module_id = module_id
+        self.stdout_queue = stdout_queue
+        self.socket_writer_queue = socket_writer_queue
+        self.socket_status_flag = socket_status_flag
+        self.log_levels_switches = [True for i in range(len(LogLevel))]
+
+    def enable_log_level(self, log_lvl_to_change):
+        try:
+            self.log_levels_switches[log_lvl_to_change.value] = True
+        except:
+            pass
+
+    def disable_log_level(self, log_lvl_to_change):
+        try:
+            self.log_levels_switches[log_lvl_to_change.value] = False
+        except:
```

Comment:
We should never catch all exceptions, instead, we should catch specifically the exception that we expect to occur.

Reply:
This applies in many places.